language/src/ceylon/language/Sequence.ceylon

```diff
@@ -287,8 +287,13 @@ shared sealed interface Sequence<out Element=Anything>
         
         first => outer.first;
         size => outer.size*times;
-        rest => sublistFrom(1).sequence(); //TODO!
-        
+        rest => if (times==1)
+                    then outer.rest
+                    else let (tail = Repeat(times-1))
+                         if (outer.longerThan(1))
+                             then outer.rest.append(tail)
+                             else tail;
+
         getFromFirst(Integer index)
```

Comment:
Not sure what's going on with GitHub... reposting:

--------

I suspect all the backends are wrong. @gavinking?

* The JVM backend incorrectly interprets the original `sublistFrom(1)` as `outer.sublistFrom(1)`, but it should be `this.sublistFrom(1)`. `sublistFrom` is `shared`, and should be inherited by `Repeat`.
* The Dart & JS backends incorrectly interpret `Repeat(times-1)` as `this.Repeat(times-1)`, but it should be `outer.Repeat(times-1)`. `Repeat` doesn't have a `Repeat` member, since `Sequence.Repeat` is not `shared`, and, the typechecker disallows `this.Repeat(...)`.

The counter point to the second item is that since `Sequence.Repeat` is visible inside `Repeat`, the following *is* allowed inside `Repeat`: `(this of Repeat).Repeat(...)`. Can the implicit `this` be implicitly widened to `Sequence` to make it a suitable receiver for `Repeat`!?...

The test code I used:

```ceylon
shared void run() {
    object a satisfies A {}
    a.testit();
}

interface A {
    shared void testit() {
        B().foo();
    }
    class B() satisfies A {
        shared void foo() {
            B(); // allowed. But what is it?
            B().printOuter(); // backends differ what's printed
            this.B(); // error (unless we make B shared)
            (B() of B).B(); // error B not defined (unless...B shared)
            (B() of A).B(); // ok
        }
        shared void printOuter() {
            print(outer);
        }
    }
}
```